### DISCUSSION TITLE
How to optimize or improve file upload performance

### DISCUSSION THREAD

**Discussion by chuanpan** [open-webui/open-webui#7179] [General]
I have observed that the file upload process is significantly taxing on CPU resources. I am currently utilizing an AWS EC2 instance (t2.large, 2 vCPUs, 8 GB RAM) to run the Open WebUI application. However, uploading five files simultaneously, each approximately 300 KB, results in a substantial slowdown, taking around three minutes to complete. During this time, CPU usage peaks at 100%, adversely affecting overall system performance. I would appreciate any guidance on optimizing this process.

**Comment by arunbugkiller** [comment]
@chuanpan facing the same issue, were you able to optimize the whole process, what content extraction engine are you using?
